frontend/src/components/TriggeredPrompt.tsx

```diff
@@ -56,113 +29,101 @@ const TriggeredPrompt: FunctionComponent<TriggeredPromptProps> = ({
   promptDelay,
   secondPromptFinishedHandler,
 }): ReactElement => {
-  let [canPlayPrompt1, setCanPlayPrompt1] = useState(false);
-  let [canPlayPrompt2, setCanPlayPrompt2] = useState(false);
-  let [prompt1Duration, setPrompt1Duration] = useState(0);
-  let [prompt2Duration, setPrompt2Duration] = useState(0);
-
-  const updatePrompt1 = (canPlay: boolean, duration: number) => {
-    setPrompt1Duration(duration);
-    setCanPlayPrompt1(canPlay);
-  };
-
-  const updatePrompt2 = (canPlay: boolean, duration: number) => {
-    setPrompt2Duration(duration);
-    setCanPlayPrompt2(canPlay);
-  };
-
-  const prompt1 = new Audio(prompt1Url);
-  const prompt2 = new Audio(prompt2Url);
-
-  prompt2.onended = () => {
-    secondPromptFinishedHandler?.();
-  };
-
-  prompt1.preload = "metadata";
-  prompt2.preload = "metadata";
-
-  useEffect(() => {
-    //loop prompt if this is an assessment
-    getDuration(prompt1, 0, 50, updatePrompt1);
-
-    if (isAssessment && canPlayPrompt1) {
-      playPrompt1();
-      const interval = setInterval(
-        playPrompt1,
-        prompt1Duration * 1000 + 8000
-      );
-      return () => {
-        stopAudio(prompt1);
-        stopAudio(prompt2);
-        clearInterval(interval);
-      };
+  const [audio1, setAudio1] = useState(new Audio(prompt1Url));
+  const audioTimeoutId = useRef<NodeJS.Timeout>();
+  const [lastPlayed, setLastPlayed] = useState<"audio1" | "audio2">(
+    "audio1"
+  );
+  const [
+    prevTriggerSecondPrompt,
+    setPrevTriggerSecondPrompt,
+  ] = useState(triggerSecondPrompt);
+  const [prevPrompt1Url, setPrevPrompt1Url] = useState(prompt1Url);
+  const [prevPrompt2Url, setPrevPrompt2Url] = useState(prompt2Url);
+
+  const [audio2, setAudio2] = useState(new Audio(prompt2Url));
+  const onClickHandler = useCallback(() => {
+    if (audioTimeoutId.current) {
+      clearTimeout(audioTimeoutId.current);
     }
-
-    // Intervention activities
-    if (prompt2Url !== undefined) {
-      getDuration(prompt2, 0, 50, updatePrompt2);
-      if (canPlayPrompt1 && canPlayPrompt2) {
-        if (triggerSecondPrompt) {
-          stopAudio(prompt1);
-          playPrompt2();
-        } else {
-          playPrompt1();
-        }
-      }
+    if (
+      audio1.paused &&
+      triggerSecondPrompt &&
+      audio2.paused &&
+      lastPlayed === "audio1"
+    ) {
+      audio1.currentTime = 0;
+      audio2.currentTime = 0;
+      audio1.pause();
+      audio2.play();
+      setLastPlayed("audio2");
+    } else if (audio1.paused && audio2.paused) {
+      audio1.currentTime = 0;
+      audio1.play();
+      setLastPlayed("audio1");
```

Comment:
so just to clarify with the logic of this if statement, if neither of these cases are true, that means an audio is already playing, so we don't want to do anything?

also I've beeen seeing the triggerSecondPrompt variable -- just curious if there's an equivalent for 'triggering' the first prompt, or is that just the default/not necessary?

Reply:
> so just to clarify with the logic of this if statement, if neither of these cases are true, that means an audio is already playing, so we don't want to do anything?

yes, this way the prompt won't be interrupted if it's already playing

> also I've beeen seeing the triggerSecondPrompt variable -- just curious if there's an equivalent for 'triggering' the first prompt, or is that just the default/not necessary?

first prompt is automatically played on page load, whereas the second prompt is only triggered in certain cases (for some activities it will play as soon as audio1 is done, while for some a button needs to be pressed first)

Reply:
also in assessments there is no second prompt